File: fsharp/space-age/SpaceAge.fs
module SpaceAge

type Planet
    = Earth 
    | Mercury
    | Venus
    | Mars
    | Jupiter
    | Saturn
    | Uranus
    | Neptune

let factor planet =
    match planet with
        | Planet.Earth -> 1m
        | Planet.Mercury -> 0.2408467m
        | Planet.Venus -> 0.61519726m
        | Planet.Mars -> 1.8808158m
        | Planet.Jupiter -> 11.862615m
        | Planet.Saturn -> 29.447498m
        | Planet.Uranus -> 84.016846m
        | Planet.Neptune -> 164.79132m

let secondsInAYear =
    factor >> (*) 31557600m 

let round (n: decimal) =
    System.Math.Round(n, 2)

let spaceAge planet seconds =
    seconds / secondsInAYear planet |> round

File: nim/space-age/space_age.nim
import math

const EARTH = 31557600

proc secondsInYear(planet: string): float =
  let factor = case planet:
    of "Earth":
      1.0
    of "Mercury":
      0.2408467
    of "Venus":
      0.61519726
    of "Mars":
      1.8808158
    of "Jupiter":
      11.862615
    of "Saturn":
      29.447498
    of "Uranus":
      84.016846
    of "Neptune":
      164.79132
    else:
      1

  EARTH * factor

proc age*(planet: string, age: int64): float =
  (float(age) / secondsInYear(planet)).round(2)